lib/daru/io/importers/mongo.rb

```diff
@@ -0,0 +1,137 @@
+        #   #  2 5948d0cecd      Volvo    29000.0
+        #
+        # @example Reading from a connection hash without JSON-path selectors
+        #
+        #   # The below 'cars' collection can be recreated in a Mongo shell with -
+        #   # db.cars.drop()
+        #   # db.cars.insert({name: "Audi", price: 52642})
+        #   # db.cars.insert({name: "Mercedes", price: 57127})
+        #   # db.cars.insert({name: "Volvo", price: 29000})
+        #
+        #   connection = { hosts: ['127.0.0.1:27017'], database: 'test' }
+        #   collection = 'cars'
+        #   Daru::IO::Importers::Mongo.new(connection, collection).call
+        #
+        #   #=> #<Daru::DataFrame(3x3)>
+        #   #           _id       name      price
+        #   #  0 5948d0bfcd       Audi    52642.0
+        #   #  1 5948d0c6cd   Mercedes    57127.0
+        #   #  2 5948d0cecd      Volvo    29000.0
+        #
+        # @example Reading from a Mongo::Client connection with JSON-path selectors
+        #
+        #   # The below 'cars' collection can be recreated in a Mongo shell with -
+        #   # db.cars.drop()
+        #   # db.cars.insert({name: "Audi", price: 52642, star: { fuel: 9.8, cost: 8.6, seats: 9.9, sound: 9.3 }})
+        #   # db.cars.insert({name: "Mercedes", price: 57127, star: { fuel: 9.3, cost: 8.9, seats: 8.4, sound: 9.1 }})
+        #   # db.cars.insert({name: "Volvo", price: 29000, star: { fuel: 7.8, cost: 9.9, seats: 8.2, sound: 8.9 }})
+        #
+        #   require 'mongo'
+        #   connection = Mongo::Client.new ['127.0.0.1:27017'], database: 'test'
+        #   collection = 'cars'
+        #   Daru::IO::Importers::Mongo.new(
+        #     connection,
+        #     collection,
+        #     '$.._id',
+        #     '$..name',
+        #     '$..price',
+        #     '$..star..fuel',
+        #     '$..star..cost'
+        #   ).call
+        #
+        #   #=> #<Daru::DataFrame(3x5)>
+        #   #          _id       name      price       fuel       cost
+        #   # 0 5948d40b50       Audi    52642.0        9.8        8.6
+        #   # 1 5948d42850   Mercedes    57127.0        9.3        8.9
+        #   # 2 5948d44350      Volvo    29000.0        7.8        9.9
+        def initialize(connection, collection, *columns, order: nil, index: nil,
+          **named_columns)
+          @client        = get_client(connection)
+          @collection    = collection.to_sym
+          @columns       = columns
+          @order         = order
+          @index         = index
+          @named_columns = named_columns
+        end
+
+        def call
+          data      = @client[@collection].find.map(&:values)
+          orders    = @client[@collection].find.map(&:keys)
+          documents = orders.map.with_index { |order, i| Hash[order.zip data[i]] }
+
+          JSON.new(
+            documents,
+            *@columns,
+            order: @order,
+            index: @index,
+            **@named_columns
+          ).call
```

Comment:
Acknowledged. I was initially apprehensive of why it'd be better for `Mongo` Importer to inherit `JSON` Importer rather than just calling it. However, requirements like raising an error when both `order` & `named_columns` were enough to justify me that inheritance would be a much DRYer way to go about it. 😄 

I've made the changes. 👍